app/controllers/spree/admin/payments_controller.rb

```diff
@@ -134,10 +134,14 @@ def load_payment
       def authorize_stripe_sca_payment
         return unless @payment.payment_method.class == Spree::Gateway::StripeSCA
 
-        @payment.authorize!
-        return if @payment.pending? && @payment.cvv_response_message.nil?
+        @payment.authorize!(spree.order_path(@payment.order, only_path: false))
 
-        raise Spree::Core::GatewayError, I18n.t('authorization_failure')
+        raise Spree::Core::GatewayError, I18n.t('authorization_failure') unless @payment.pending?
+
+        return unless @payment.cvv_response_message.present?
+
+        PaymentMailer.authorize_payment(@payment).deliver
```

Comment:
I think the `ActionMailer#deliver` method is now deprecated (we still use it in various places, but we should probably update it). Either `#deliver_now` (direct equivalent) or `#deliver_later` should be used.

This makes me wonder: would it be better to use `#deliver_later` in these cases (especially where the order confirmation is happening)? Basically sending of the email will be handled asyncronously via delayed job, which means a) it's not blocking the thread and b) if there's an error with sending, it can be retried automatically.

What do you think?

Reply:
Yep, I think I'd just seen `#deliver` used elsewhere so I assumed that was the preference. In this case I think `#deliver_later` is appropriate 

ec4398e5702403175d00355c92d83f6bb3e2c2df